rotkehlchen/chain/evm/decoding/gearbox/decoder.py

```diff
@@ -211,8 +226,62 @@ def _decode_pool_events(self, context: DecoderContext) -> DecodingOutput:
 
         return DEFAULT_DECODING_OUTPUT
 
+    def _decode_stake(self, context: DecoderContext) -> DecodingOutput:
+        user_address = hex_or_bytes_to_address(context.tx_log.topics[1])
+        amount = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[:32]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,
+        )
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == amount and
+                event.location_label == user_address
+            ):
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.counterparty = CPT_GEARBOX
+                event.product = EvmProduct.STAKING
+                event.notes = f'Stake {amount} GEAR'
+                break
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_unstake(self, context: DecoderContext) -> DecodingOutput:
+        user_address = hex_or_bytes_to_address(context.tx_log.data[:32])
+        amount = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[32:64]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,
+        )
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == amount and
+                event.location_label == user_address
+            ):
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.counterparty = CPT_GEARBOX
+                event.product = EvmProduct.STAKING
+                event.notes = f'Unstake {amount} GEAR'
+                break
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_staking_events(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] == DEPOSIT_GEAR:
+            return self._decode_stake(context=context)
+
+        if context.tx_log.topics[0] == CLAIM_GEAR_WITHDRAWAL:
+            return self._decode_unstake(context=context)
+
+        return DEFAULT_DECODING_OUTPUT
+
     def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
-        return dict.fromkeys(self.pools, (self._decode_pool_events,))
+        return dict.fromkeys(self.pools, (self._decode_pool_events,)) | {self.staking_contract: (self._decode_staking_events,)}  # noqa: E501
```

Comment:
And again making my point on this system being unsustainable. Since you are adding an `EvmProduct` here you should add this to the decoder: https://github.com/rotki/rotki/blob/4b5d2ebaabeeb97958b3c587d081f477993688a7/rotkehlchen/chain/ethereum/modules/convex/decoder.py#L276

Reply:
Added